pkg/database/dboptions.go

```diff
@@ -36,14 +41,21 @@ type Options struct {
 	corruptionChecker bool
 
 	readTxPoolSize int
+
+	// TruncationFrequency determines how frequently to truncate data from the database.
+	TruncationFrequency time.Duration
+
+	// RetentionPeriod determines how long to store data in the database.
+	RetentionPeriod time.Duration
 }
 
 // DefaultOption Initialise Db Optionts to default values
 func DefaultOption() *Options {
 	return &Options{
-		dbRootPath:     DefaultDbRootPath,
-		storeOpts:      store.DefaultOptions(),
-		readTxPoolSize: DefaultReadTxPoolSize,
+		dbRootPath:          DefaultDbRootPath,
+		storeOpts:           store.DefaultOptions(),
+		readTxPoolSize:      DefaultReadTxPoolSize,
+		TruncationFrequency: DefaultTruncationFrequency,
```

Comment:
others attributes are private, does it require to be public?

Reply:
yes, because this is being used by the truncator to know the trunc frequency and retention period